web/app/js/components/Version.jsx

```diff
@@ -0,0 +1,76 @@
+import corslite from '@mapbox/corslite';
+import { Link } from 'react-router-dom';
+import React from 'react';
+import './../../css/version.css';
+
+export default class Version extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.handleResponse = this.handleResponse.bind(this);
+    this.state = {err: null, done: false, latest: null};
+  }
+
+  componentDidMount() {
+    this.loadFromServer();
+  }
+
+  handleResponse(error, resp) {
+    let latest = null;
+    let err = null;
+
+    if (resp && resp.responseText) {
+      let json = JSON.parse(resp.responseText);
+      if (json.version) {
+        latest = json.version;
+      }
+    }
+
+    if (!latest && error && error.statusText) {
+      err = error.statusText;
+    }
+
+    this.setState({ err: err, done: true, latest: latest });
+  }
+
+  loadFromServer() {
+    if (this.state.done) {
+      return;
+    }
+    corslite(
+      `https://versioncheck.conduit.io/version?version=${this.props.releaseVersion}?uuid=${this.props.uuid}`,
+      this.handleResponse,
+      true
+    );
+  }
+
+  renderVersionCheck() {
+    if (!this.state.done) {
+      return "Performing version check...";
+    }
+
+    if (!this.state.latest) {
+      return (<div>
+        Version check failed
+        {this.state.err? ": "+this.state.err : null}
+      </div>);
+    } else if (this.state.latest === this.props.releaseVersion) {
+      return "Conduit is up to date";
+    } else {
+      return (<div>
+        A new version ({this.state.latest}) is available<br />
```

Comment:
I don't know anything about react. What ensures that this is safe w.r.t. xss? I.e. let's say `this.state.latest` is `"<script>alert("hi");</script>"`, does this syntax do the escaping to make that safe?

Reply:
Good question!

I modified my test response to be:
```json
{
  "version": "<script>alert('hi');</script>"
}
```

...and got:

![screen shot 2017-12-22 at 3 39 15 pm](https://user-images.githubusercontent.com/236915/34314933-573fcc20-e72e-11e7-9641-62c1f6a282b7.png)

Is that sufficient? If there's a more malicious way to craft that json, I can try it pretty quickly.

Reply:
And you're correct @briansmith -- the `{...}` syntax has built in escaping, documented here: https://reactjs.org/docs/jsx-in-depth.html#string-literals-1

Reply:
@siggy @klingerf Thanks! I agree that's what's wanted.